[Merged by Bors] - feat: split Topology/Connected.lean

grunweg opened the pull request:
In the last step, I have removed *redundant* imports: those which are implied by the other imports.
I can revert those changes if desired/if this seems too brittle.

fpvandoorn commented on Mathlib/Topology/Connected/LocallyConnected.lean:
```diff
@@ -0,0 +1,150 @@
+of connected (not necessarily open) sets --- but in a non-trivial way, so we choose this definition
+and prove the equivalence later in `locallyConnectedSpace_iff_connected_basis`.
+-/
+-- FIXME: check copyright
```
Could you do this? Otherwise LGTM.

semorrison replied:
I tend to just copy the full authors list from a "parent" file to each "child" file when splitting. It's an over-approximation for both credit and blame (=asking for help when something goes wrong), but for both over-approximation seems better than under.

fpvandoorn commented:
Moving PathConnected into the new folder sounds good, and `Connected` is a better name than `Connectedness`.

grunweg commented:
I have addressed all comments and rebased on top of latest master.

For the authors and copyright lines.
Local connectedness was added [in 2022](https://github.com/leanprover-community/mathlib/commit/e1f01165ce7a8b650014e98bab47cdff5317e52e) in a PR by Anatole Dedecker, upstreaming results from the sphere eversion project. I've put them as copyright holder; let me know if something else is preferred.

For totally disconnected and totally separated spaces, code archeology is more interesting: after a handful of file splits, lots of refactors etc., git blame [eventually arrives](https://github.com/leanprover-community/mathlib/blame/dcd0466de4add821402454281552eeaef0593843/analysis/topology/topological_space.lean) at [Kenny Lau's commit in 2018](https://github.com/leanprover-community/mathlib/commit/849d2a41a46f58a7abfe9f03b7966d245ef9db8e). This commit didn't add all of today's content, but contains the basic definitions and properties.

Most material was added later, spread over several commits, including [upstreaming LTE lemmas](https://github.com/leanprover-community/mathlib/commit/6d7d16988b660a6e1bf4b2475039d3b677d776e1), [Calle Sönne](https://github.com/leanprover-community/mathlib/pull/6188 Calle Sönne) and [laughinggas](https://github.com/leanprover-community/mathlib/commit/89928bc0d44f82f3b5546632c46ee495ee3389b1). I don't find an obvious answer for whom to list as author; I made an approximate guess among people still around.

fpvandoorn commented:
Appoximations are good (and necessary - we don't add everyone as an author).

Thanks for the fixes!

bors merge

One remark for next time: after you've received your first review, it's preferred that you don't force push anymore. This makes it easier to see for the reviewers what has changed since their last review. So use `git merge` instead of `git rebase`. (In this case it wasn't a big deal, since it was clear what commits you added after my review.)

grunweg commented:
> Appoximations are good (and necessary - we don't add everyone as an author).
> 
> Thanks for the fixes!
> 
> bors merge
> 
> One remark for next time: after you've received your first review, it's preferred that you don't force push anymore. This makes it easier to see for the reviewers what has changed since their last review. So use `git merge` instead of `git rebase`. (In this case it wasn't a big deal, since it was clear what commits you added after my review.)

Noted, thanks for reminding me. (I generally try to obey this. In this case, I noticed a trivial error in the commit I just pushed; I amended that commit instead of pushing a new "oops" commit. Old habits die hard...)
